src/components/review/ReviewsList.js

```diff
@@ -0,0 +1,47 @@
+import React, { useEffect, useState } from 'react';
+import axiosInstance from '../../apis/axiosClient';
+
+const AlbumReviews = ({ albumId }) => {
+  const [reviews, setReviews] = useState([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    // Fetch reviews for the given album
+    async function fetchAlbumReviews() {
+      try {
+        const response = await axiosInstance.get(`/reviews/album/${albumId}`);
+        console.log(response.data.allProductReviews);
+
+        const { allProductReviews } = response.data;
+        setReviews(allProductReviews);
+        setLoading(false);
+      } catch (error) {
+        console.error('Error fetching album reviews:', error);
+        setLoading(false);
+      }
+    }
+
+    fetchAlbumReviews();
+  }, [albumId]);
+
+  return (
+    <div>
+      <h2>Album Reviews</h2>
+      {loading ? (
+        <p>Loading reviews...</p>
+      ) : (
+        <ul>
+          {reviews.map((review) => (
+            <li key={review._id}>
+              <h3>{review.title}</h3>
+              <p>Rating: {review.rating}</p>
+              <p>{review.comment}</p>
+            </li>
+          ))}
+        </ul>
+      )}
+    </div>
```

Comment:
I would rather here do:
`{ loading && <p>Loading reviews...</p>}`

and then as a separate check, we can either list the reviews or show the empty state:

`{ reviews?.length ? ( <ul> ....etc </ul> ) : "No reviews yet" }`

One nice thing is that means all our code for loading, populated, and empty state lives in the same component instead of the empty state being all the way in `AlbumPreview.js` as it is right now. 

---

Another approach that some prefer is to pull the conditionals into their own render code instead of having lots of nested ternaries inside the one return statement. This could look like:

```js
const AlbumReviews = ({ albumId }) => {
 // all our state and use effect stuff is still here, I'm just leaving it out for the sake of space

if (loading) return <p>Loading reviews...</p>
if (!reviews?.length) return <p>No reviews yet</p>

return (
  // reviews.map etc
)
```

Reply:
@akosasante thank you, just deleted  this state message 'No reviews' from  AlbumPreview.js is that correct?
![image](https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/103755090/07a92f01-34ef-464b-9b98-0c919d47dbc9)

Reply:
Yes; currently both places that use `<AlbumPreview>` use it and pass in `children` (nested components) but if we want this component to be useable without that, then maybe we should keep the `{children && <div style={{...}}>{children}</div>` part at least; so that the div for the children is only rendered if any children were actually passed in.